client/pages/product/viewproduct.jsx

```diff
@@ -1,11 +1,13 @@
 import React, { useState, useEffect } from 'react';
 import { axiosInstance } from '../../configs/axios';
 import styles from '../../styles/viewproduct.module.css'; // Import the CSS file
+import UpdateProduct from './updateproduct'; // Import the UpdateProduct component
 
 const ViewProducts = () => {
   const [products, setProducts] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
+  const [selectedProduct, setSelectedProduct] = useState(null); // State to store the selected product for update
 
   useEffect(() => {
     const fetchProducts = async () => {
```

Comment:
You're using tokens from localstorage here again, this is incorrect, you should refactor all the usage when it was reported for the first time.